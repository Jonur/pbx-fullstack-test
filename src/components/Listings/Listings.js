import React from 'react';
import { array } from 'prop-types';

const Listings = ({ listings }) => (
  <section className="pb-section">
    <h1 className="section-title">Listings</h1>
    {
      listings.map((product, index) => (
        <div data-test="product-listing" key={index}>
          Product
        </div>
      ))
    }
  </section>
);

Listings.propTypes = {
  listings: array.isRequired,
};

export default Listings;