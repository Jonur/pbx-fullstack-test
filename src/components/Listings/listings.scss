$tablet-width: 480px;
$desktop-width: 768px;

@mixin tablet {
  @media (min-width: #{$tablet-width}) {
    @content;
  }
}

@mixin desktop {
  @media (min-width: #{$desktop-width}) {
    @content;
  }
}

.pb-section {
  font-size: 1rem;
  width: 100%;
  max-width: 1024px;
  margin: 0 auto;
  color: #3a4155;

  @include tablet {
    font-size: 1rem;
  }


  .section-title {
    font-size: 2rem;
    text-align: center;
  }

  .products {
    display: flex;
    flex-flow: row wrap;
    justify-content: space-around;

    @include tablet {
      align-content: center;
    }


    > div {
      flex: 1 100%;
      text-align: center;
      border: 2px solid #dfe0ed;
      padding: 20px;
      margin-bottom: 20px;

      @include tablet {
        flex: 1 1 30%;
        max-width: 30%;
      }


      @include desktop {
        flex: 1;
        margin-right: 10px;
        max-width: 28%;
      }


      .product-cover {
        max-width: 100%;
      }

      .product-labels {
        height: 30px;
        min-height: 30px;

        .product-label {
          margin: 0 -20px;
          padding: 5px 20px;
          text-align: left;
          color: white;
          font-weight: bold;
          background-color: #ff2c13;
        }
      }

      .product-title, .product-labels, p, .price {
        text-align: left;
      }

      p {
        margin: 30px 0;
      }

      .price {
        > span:first-of-type {
          margin-right: 10px;
          font-size: 1rem;

          @include tablet {
            font-size: 2rem;
          }
        }

        > span:last-of-type {
          font-weight: bold;
          font-size: 2rem;

          @include tablet {
            font-size: 3rem;
          }
        }
      }

      .product-cta {
        display: block;
        border: 5px solid #00ff00;
        margin: 20px auto;
        padding: 20px;
        font-size: 1.5rem;
        font-weight: bold;

        @include tablet {
          font-size: 2rem;
        }
      }

      .product-cta, .product-cta:hover, .product-cta:visited {
        color: #3a4155;
        text-decoration: none;
      }
    }
  }
}
