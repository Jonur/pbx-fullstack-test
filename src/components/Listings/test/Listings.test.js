import React from 'react';
import Listings from '../Listings';

describe('Listings Component', () => {
  const defaultProps = {
    listings: [
      {
        'title': 'Simple Canvas',
        'description': 'Lets your pictures speak for themselves.',
        'image': {
          'path': '/images/product.jpg',
          'alt': 'Simple Canvas'
        },
        'price': 1500,
        'currency': '£',
        'priceLabel': 'From',
        'productLabel': 'bestseller',
        'cta': 'Shop Now',
        'ctaLink': '/random/link/to/no/where'
      },
      {
        'title': 'Collage Canvas',
        'description': 'Can\'t choose just one pic? Put your favourite photos on one canvas - a collage.',
        'image': {
          'path': '/images/product.jpg',
          'alt': 'Collage Canvas'
        },
        'price': 2500,
        'currency': '£',
        'priceLabel': 'From',
        'productLabel': '',
        'cta': 'Shop Now',
        'ctaLink': '/random/link/to/no/where'
      },
    ],
  };

  const wrapper = shallow(<Listings {...defaultProps} />);

  it('should render the component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should contain 2 product listing cards', () => {
    expect(wrapper.find('[data-test="product-listing"]')).toHaveLength(2);
  });
});