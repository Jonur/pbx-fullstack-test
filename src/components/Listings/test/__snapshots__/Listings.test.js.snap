// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Listings Component should render the component 1`] = `
<section
  className="pb-section"
>
  <h1
    className="section-title"
  >
    Listings
  </h1>
  <div
    data-test="product-listing"
    key="0"
  >
    Product
  </div>
  <div
    data-test="product-listing"
    key="1"
  >
    Product
  </div>
</section>
`;
